### ISSUE TITLE
[Question] 源码编译失败

### ISSUE THREAD

**Issue by zerotocode398** [dataease/dataease#6257]
**请描述您的问题.**
执行mvn clean package -Pstage编译失败
```
[root@dataease backend]# pwd
/home/tianciwang/dataease/core/backend
[root@dataease backend]# mvn clean package -Pstage
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for io.dataease:backend:jar:1.18.11
[WARNING] 'version' contains an expression but should be a constant. @ io.dataease:dataease-server:${dataease.version}, /home/tianciwang/dataease/core/pom.xml, line 7, column 14
[WARNING] 'version' contains an expression but should be a constant. @ io.dataease:dataease:${dataease.version}, /home/tianciwang/dataease/pom.xml, line 8, column 14
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] 
[INFO] ------------------------< io.dataease:backend >-------------------------
[INFO] Building backend 1.18.11
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for pentaho-kettle:kettle-core:jar:8.3.0.18-1112 is missing, no dependency information available
[WARNING] The POM for pentaho-kettle:kettle-engine:jar:8.3.0.18-1112 is missing, no dependency information available
[WARNING] The POM for pentaho:metastore:jar:8.3.0.18-1112 is missing, no dependency information available
[WARNING] The POM for org.pentaho.di.plugins:pdi-engine-configuration-impl:jar:8.3.0.7-683 is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.500 s
[INFO] Finished at: 2023-10-11T16:28:57+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project backend: Could not resolve dependencies for project io.dataease:backend:jar:1.18.11: Failed to collect dependencies at io.dataease:dataease-plugin-interface:jar:1.18.11: Failed to read artifact descriptor for io.dataease:dataease-plugin-interface:jar:1.18.11: io.dataease:dataease:pom:1.18.11 was not found in https://maven.aliyun.com/repository/public during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of aliyunmaven has elapsed or updates are forced -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[root@dataease backend]# mvn -v
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /opt/apache-maven-3.8.4
Java version: 1.8.0_382, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.382.b05-1.el7_9.x86_64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.10.0-1160.el7.x86_64", arch: "amd64", family: "unix"
[root@dataease backend]# node -v
v16.13.0
```

使用的是v1.18.11 tag，commit是
```
[root@dataease backend]# git log | head -2
commit 5f3a33acceac41b158735f405e8263f99c93bdf7
Merge: f677880 5063707
```

**Comment by siriusxxk21** [comment]
遇到了同样的问题

**Comment by zerotocode398** [comment]
> 遇到了同样的问题
用容器运行，把里面编译好的backen.jar拷贝出来

**Comment by tjlygdx** [comment]
现在本地安装一下依赖
cd sdk && mvn clean install

**Comment by ysy1418899560** [comment]
在本地安装依赖之后，依旧报错

**Comment by zerotocode398** [comment]
> 现在本地安装一下依赖 cd sdk && mvn clean install

不行，还是这个错，怎么处理，急急急

**Comment by yimogit** [comment]
我是 v1.18.12

这是我的报错
![image](https://github.com/dataease/dataease/assets/15975059/9cc804e1-f30d-426d-946c-60fb5c0fcdf8)

我是在database 和 sdk目录都执行 `mvn clean install`后，确保mvn本地仓库有1.18.12的包
![image](https://github.com/dataease/dataease/assets/15975059/9a96019d-0c3a-47e0-b592-ec02e499bff5)

修改backend\pom.xml的依赖指定版本

```
        </dependency>
        <dependency>
            <groupId>io.dataease</groupId>
            <artifactId>dataease-plugin-interface</artifactId>
            <version>1.18.12</version>
            <!-- <exclusions>
                <exclusion>
                    <artifactId>guava</artifactId>
                    <groupId>com.google.guava</groupId>
                </exclusion>
            </exclusions> -->
        </dependency>
```

再执行编译：`mvn clean package` 成功运行
![image](https://github.com/dataease/dataease/assets/15975059/24fdfeee-9057-4306-9c30-f085c27449ff)

**Comment by xuwei-fit2cloud** [comment]
mvn clean install -N

echo "---------------------编译SDK---------------------"
cd sdk && mvn clean install

echo "---------------------编译core---------------------"
cd ../core && mvn clean package -U -Dmaven.test.skip=true

**Comment by huguiqi** [comment]
其它全部都OK，就core里面的backend报错，说缺依赖：Could not resolve dependencies for project io.dataease:backend:jar:1.18.17: The following artifacts could not be resolved: pentaho-kettle:kettle-core:jar:8.3.0.18-1112, pentaho-kettle:kettle-engine:jar:8.3.0.18-1112, pentaho:metastore:jar:8.3.0.18-1112, org.pentaho.di.plugins:pdi-engine-configuration-impl:jar:8.3.0.7-683: Could not find artifact pentaho-kettle:kettle-core:jar:8.3.0.18-1112 in pentaho-public (https://repo.orl.eng.hitachivantara.com/artifactory/pnt-mvn/) -> [Help 1]

**Comment by huguiqi** [comment]
原来的那个仓库已经不可用了，换成这个仓库：https://repo.orl.eng.hitachivantara.com/artifactory/pnt-mvn/ 后，需要将pentaho-kettle:kettle-core:jar的版本号改为：9.3.0.0-428,就好了，国内开源做的还是不够
